starknet-crypto/src/with_std.rs

```diff
@@ -0,0 +1,6 @@
+pub mod with_std {
+    pub use std::{fmt, ops, str};
```

Comment:
The global `core` crate is available even when `no_std` is off (aka. normal mode). So we should not need this conditional import? Should be able to just change all `std::fmt` etc. to `core::fmt` unconditionally for example?

Reply:
I think the best approach here is to apply the clippy lint(s) project-wide in `master` first, and rebase this PR? That way we don't have to manually inspect every conditional import to see if they're not needed.

Reply:
Played around the idea a bit and got bitten by `thiserror` lol - enabling the lints results in every `thiserror::Error` derive getting error. So I would have to deal with `thiserror` first, which is exactly what this PR already does.

So I'll simply add the lints on top of this PR to `starknet-core` and its dependencies to make sure we're not doing unnecessary conditional imports. Then I will apply the lints to other crates in a separate PR.

(btw I'm kinda annoyed that there's no way to enable/disable lints globally in a workspace/project, not until this merges: https://github.com/rust-lang/rfcs/pull/3389)

Reply:
> Should be able to just change all std::fmt etc. to core::fmt unconditionally for example?

Yes we can. But then we have to do conditional import for symbols in `alloc`:
```rust
#[cfg(feature = "std")]
use std::rc::Rc;
#[cfg(all(not(feature = "std")), feature = "alloc")]
use alloc::rc::Rc;
```

I tried both and I really find it more comfortable to have all those stuffs managed once and for all in one place, by the library maintainer, and then anybody contributing just has to import what it needs from one single place.

It may seem a bit artificial for small repositories that don't do extensive usage of the `std`/`core`/`alloc` symbols, but in bigger repositories, it is such a quality-of-life improvement.

Nevertheless, I'm open to removing it and keeping it manual if it really bother you

Reply:
About `thiserror`. It does not support `no_std`. I used `this-error-no-std` instead, which is kind of a wrapper crate I believe. With a lot of complex macro in it, so it is possible that it does strange stuffs to the linter
`snafu` is an alternative crate that supports `no_std` natively, but it is not iso-feature with `thiserror`. It's another paradigm.